server/src/service/collection.rs

```diff
@@ -0,0 +1,201 @@
+// org.freedesktop.Secret.Collection
+
+use std::collections::HashMap;
+
+use oo7::dbus::api::properties::Properties;
+use oo7::portal::Item;
+use serde::{Deserialize, Serialize, Serializer};
+use zbus::{dbus_interface, zvariant, Error, ObjectServer, SignalContext};
+use zvariant::{ObjectPath, OwnedObjectPath, Signature, Type};
+
+use crate::service::item;
+use crate::KEYRING;
+
+#[derive(Default, Debug, Deserialize)]
+pub struct Collection {
+    pub items: Vec<item::Item>,
+    label: String,
+    locked: bool,
+    created: u64, // how to represent date
+    modified: u64,
```

Comment:
Can't those use std types? see the client library

Reply:
I tried replacing `u64` with `std::time::Duration` and also returning `u64`s as `std::time::Duration`s in getters. Both times I'm getting the following error,
```
error[E0277]: the trait bound `zbus::zvariant::Structure<'_>: From<Duration>` is not satisfied
........
#[dbus_interface(name = "org.freedesktop.Secret.Collection")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Duration>` is not implemented for `zbus::zvariant::Structure<'_>`
.........
```